packages/api-server/src/__tests__/lambdaLoader.test.ts

```diff
@@ -0,0 +1,73 @@
+import path from 'path'
+
+import {
+  LAMBDA_FUNCTIONS,
+  loadFunctionsFromDist,
+} from '../plugins/lambdaLoader'
+
+// Suppress terminal logging.
+console.log = jest.fn()
+console.warn = jest.fn()
+
+// Set up RWJS_CWD.
+let original_RWJS_CWD
+
+beforeAll(() => {
+  original_RWJS_CWD = process.env.RWJS_CWD
+
+  process.env.RWJS_CWD = path.resolve(__dirname, 'fixtures/redwood-app')
+})
+
+afterAll(() => {
+  process.env.RWJS_CWD = original_RWJS_CWD
+})
+
+// Reset the LAMBDA_FUNCTIONS object after each test.
+afterEach(() => {
+  for (const key in LAMBDA_FUNCTIONS) {
+    delete LAMBDA_FUNCTIONS[key]
+  }
+})
+
+describe('loadFunctionsFromDist', () => {
+  it('loads functions from the api/dist directory', async () => {
+    expect(LAMBDA_FUNCTIONS).toEqual({})
+
+    await loadFunctionsFromDist()
+
+    expect(LAMBDA_FUNCTIONS).toEqual({
+      env: expect.any(Function),
+      graphql: expect.any(Function),
+      health: expect.any(Function),
+      hello: expect.any(Function),
+      nested: expect.any(Function),
+    })
+  })
+
+  // We have logic that specifically puts the graphql function at the front.
+  it('puts the graphql function first', async () => {
+    expect(LAMBDA_FUNCTIONS).toEqual({})
+
+    await loadFunctionsFromDist()
+
+    expect(Object.keys(LAMBDA_FUNCTIONS)[0]).toEqual('graphql')
```

Comment:
This comment isn't specifically about this test so sorry in advance. I'm happy with the test. 

We really can't rely on our current code to have the 'graphql' function appear first in the list of object keys like this. I can't find a direct link to any official specification but please see [this stackoverflow answer](https://stackoverflow.com/a/5525820) and [this blog post](https://2ality.com/2015/10/property-traversal-order-es6.html) for a quick overview. Object key order does not behave the same as an array where you can explicitly set the order.

You can see this breakdown when you add a function called `2.js` which becomes the first result regardless of our code. I know that might be a little abstract but users are free to name the functions as they please. 

I don't have the full context of why this reordering code exists. If this was done for performance I suggest we should revisit this, document the results for various testing and then update the implementation. If it is for performance I would bet this is going to be very nuanced.

Reply:
I don't have context either, but yeah you're right, nice find! I'll add a test for functions named with numbers till we get context.

Reply:
Added in https://github.com/redwoodjs/redwood/pull/9325/commits/c2df81793d4a4aeda0a44ec18b85147c607284e3 and I'll ask for context this week